### ISSUE TITLE
TiCDC keeps warning on removed changefeeds

### ISSUE THREAD

**Issue by lichunzhu** [pingcap/ticdc#2156]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
Set up TiCDC, start a changefeed, and then remove it forcely.

2. What did you expect to see?
Prometheus alerts won't be triggered any more.

3. What did you see instead?
cdc_checkpoint_high_delay
cdc_processor_checkpoint_tso_no_change_for_1m
cdc_resolvedts_high_delay
tikv_cdc_min_resolved_ts_no_change_for_1m
still alerts.

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - TiCDC version (execute `cdc version`):

        ```
        (paste TiCDC version here)
        ```
        v5.0.1,v4.0.12

**Comment by lichunzhu** [comment]
Although prometheus metrics is deleted, we should also adjust alert rules.

**Comment by lichunzhu** [comment]
more details:
![image](https://user-images.githubusercontent.com/21104942/123372597-6fb79b00-d5b6-11eb-908e-c5171ae869cb.png)
![image](https://user-images.githubusercontent.com/21104942/123372758-be653500-d5b6-11eb-966b-5b7fcc6ca256.png)

**Comment by amyangfei** [comment]
I have checked in a test environment,
- where I created two changedfeeds, naming `bypass-cf-1` and `bypass-cf-2`
- after running some time, I removed the changefeed `bypass-cf-2`

- It is to observe that metrics of `bypass-cf-2` stop updating, as follows

<img width="1368" alt="Screen Shot 2021-06-26 at 23 07 59" src="https://user-images.githubusercontent.com/1527315/123517415-6112d600-d6d3-11eb-8fd8-16b52e4afbaa.png">

- The given alert doesn't fire in prometheus, for example, the expr for alert item `changes(ticdc_processor_checkpoint_ts[1m])` is `expr:changes(ticdc_processor_checkpoint_ts[1m]) < 1`, we can check the expression in prometheus and only metric from `bypass-cf-1` is available

<img width="1158" alt="Screen Shot 2021-06-26 at 23 10 20" src="https://user-images.githubusercontent.com/1527315/123517469-b18a3380-d6d3-11eb-9aef-628066a86b82.png">

**Comment by amyangfei** [comment]
Closed by #2177 and #23133

**Comment by 3pointer** [comment]
tikv should handle `tikv_cdc_min_resolved_ts_no_change_for_1m` alert.
